Ginger/Ginger/AnalyzerLib/AnalyzerPage.xaml.cs

```diff
@@ -309,111 +309,129 @@ public void ReportUnusedVariables(object obj, List<string> usedVariables)
         {
-            AnalyzeAction AA = (AnalyzeAction)sender;
-
-            if (AA.Details.Equals("Activity"))
+            if (sender.GetType().Equals(typeof(AnalyzeActivity)))
             {
-                Activity activity = AA.mActivity;
+                Activity activity = ((AnalyzeActivity)sender).mActivity;
                 foreach (VariableBase var in activity.Variables)
                 {
-                    if (var.Name.Equals(AA.ItemName))
+                    if (var.Name.Equals(((AnalyzeActivity)sender).ItemName))
                     {
                         activity.Variables.Remove(var);
                         activity.RefreshVariablesNames();
-                        AA.Status = eStatus.Fixed;
+                        ((AnalyzeActivity)sender).Status = eStatus.Fixed;
+                        break;
                     }
                 }
             }
-            else
+            else if(sender.GetType().Equals(typeof(AnalyzeBusinessFlow)))
             {
-                BusinessFlow BFlow= AA.mBusinessFlow;                
-                if (AA.Details.Equals("BusinessFlow"))
-                {                    
-                    foreach (VariableBase var in BFlow.Variables)
+                BusinessFlow BFlow= ((AnalyzeBusinessFlow)sender).mBusinessFlow;                                                   
+                foreach (VariableBase var in BFlow.Variables)
+                {
+                    if (var.Name.Equals(((AnalyzeBusinessFlow)sender).ItemName))
                     {
-                        if (var.Name.Equals(AA.ItemName))
-                        {
-                            BFlow.Variables.Remove(var);
-                            AA.Status = eStatus.Fixed;                            
-                        }
+                        BFlow.Variables.Remove(var);
+                        ((AnalyzeBusinessFlow)sender).Status = eStatus.Fixed;
+                        break;
                     }
                 }
-                else
-                {                 
-                    foreach (VariableBase var in BusinessFlow.SolutionVariables)
+            }
+            else if (sender.GetType().Equals(typeof(AnalyzeSolution)))
+            {                 
+                foreach (VariableBase var in BusinessFlow.SolutionVariables)
+                {
+                    if (var.Name.Equals(((AnalyzeSolution)sender).ItemName))
                     {
-                        if (var.Name.Equals(AA.ItemName))
-                        {
-                            BusinessFlow.SolutionVariables.Remove(var);
-                            AA.Status = eStatus.Fixed;                                                                                    
-                        }
+                        BusinessFlow.SolutionVariables.Remove(var);
+                        ((AnalyzeSolution)sender).Status = eStatus.Fixed;
+                        break;
```

Comment:
why we have all of this analyze code inside AnalyzerPage? the analyze need to be inside the items itself like ActivityAnalyzer/ BusinessFlowAnalyzer/... what i'm missing?